src/main/java/edu/ucdavis/dss/ipa/services/jpa/JpaTeachingCallReceiptService.java

```diff
@@ -231,30 +231,39 @@ public TeachingCallReceipt create(TeachingCallReceipt teachingCallReceipt) {
 	}
 
 	@Override
-	public List<TeachingCallReceipt> createMany(List<Long> instructorIds, TeachingCallReceipt teachingCallReceiptDTO) {
+	public List<TeachingCallReceipt> createOrUpdateMany(List<Long> instructorIds, TeachingCallReceipt teachingCallReceiptDTO) {
 		List<TeachingCallReceipt> receipts = new ArrayList<>();
 
 		for (Long instructorId : instructorIds) {
+			TeachingCallReceipt teachingCallReceipt = this.findByInstructorIdAndScheduleId(instructorId, teachingCallReceiptDTO.getSchedule().getId());
+
+			if (teachingCallReceipt == null) {
+				teachingCallReceipt = new TeachingCallReceipt();
+			}
+
 			Instructor slotInstructor = instructorService.getOneById(instructorId);
-			TeachingCallReceipt slotTeachingCallReceipt = new TeachingCallReceipt();
 
-			slotTeachingCallReceipt.setSchedule(teachingCallReceiptDTO.getSchedule());
-			slotTeachingCallReceipt.setInstructor(slotInstructor);
-			slotTeachingCallReceipt.setIsDone(false);
-			slotTeachingCallReceipt.setMessage(teachingCallReceiptDTO.getMessage());
-			slotTeachingCallReceipt.setNextContactAt(teachingCallReceiptDTO.getNextContactAt());
-			slotTeachingCallReceipt.setShowUnavailabilities(teachingCallReceiptDTO.getShowUnavailabilities());
-			slotTeachingCallReceipt.setTermsBlob(teachingCallReceiptDTO.getTermsBlob());
-			slotTeachingCallReceipt.setDueDate(teachingCallReceiptDTO.getDueDate());
+			teachingCallReceipt.setSchedule(teachingCallReceiptDTO.getSchedule());
```

Comment:
Move .setSchedule() and .setInstructor() to be inside the if block on line 240 as we don't want to change those values should the teachingCallReceipt have been found on line 238.